packages/server/graphql/public/types/ReflectTemplate.ts

```diff
@@ -64,17 +46,39 @@ const ReflectTemplate: ReflectTemplateResolvers = {
       .flat()
       .map((team) => team.id)
 
-    const dateLastUsedForOrg = await getLastUsedDateForTeams(
-      orgTeamIds.filter((orgId) => !authToken.tms.includes(orgId)),
-      id,
-      dataLoader
+    // The goal here is to issue all fetches to the DB at once, then we can parse out team membership
+    const orgTeamMeetingTemplates = (
+      await Promise.all(
+        orgTeamIds.map((teamId) => dataLoader.get('teamMeetingTemplateByTeamId').load(teamId))
+      )
+    ).flat()
+
+    const orgTeamMeetingTemplatesForTemplateId = orgTeamMeetingTemplates.filter(
+      (tmt) => tmt.templateId === id
     )
-    if (dateLastUsedForOrg > new Date(Date.now() - ms('30d'))) {
+
+    const lastUsedAtOnTeam = Math.max(
+      ...orgTeamMeetingTemplatesForTemplateId
+        .filter((row) => authToken.tms.includes(row.teamId))
+        .map((row) => row.lastUsedAt.getTime())
+    )
+
+    if (lastUsedAtOnTeam > Date.now() - ms('30d')) {
+      subCategories.push('recentlyUsed')
+    }
+
+    const lastUsedAtOnOrg = Math.max(
+      ...orgTeamMeetingTemplatesForTemplateId
+        .filter((row) => !authToken.tms.includes(row.teamId))
+        .map((row) => row.lastUsedAt.getTime())
+    )
+
+    if (lastUsedAtOnOrg > Date.now() - ms('30d')) {
       subCategories.push('recentlyUsedInOrg')
     }
```

Comment:
+1 Consider pulling a lot of this back into the `getLastUsedDateForTeams` helper? I _think_ we could move [these lines](https://github.com/ParabolInc/parabol/pull/8708/files#diff-15767a301911509b2ae53ecd359954a07080796e808d229142a243825450fae8R49-R64) into that helper, and then do
```
const [lastUsedAtOnTeam, lastUsedAtOnOrg] = await Promise.All([
  getLastUsedDateForTeams(authToken.tms, id, dataLoader),
  getLastUsedDateForTeams(
      orgTeamIds.filter((orgId) => !authToken.tms.includes(orgId)),
      id,
      dataLoader
    )
])
```
and still have DataLoader batch it into a single DB call (I haven't tested this though).

Reply:
done!